### PR TITLE
Tighten type ignores

### PR DESCRIPTION
**Pull request by mr-c**
## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: tools/next_gen_conversion/solid2fastq.py [hunk 1]
```diff
@@ -1,15 +1,10 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Comment (nsoranzo) [inline tools/next_gen_conversion/solid2fastq.py]: We can't do that without adding a `<requirement />` for some version of Python 3 to `tools/next_gen_conversion/solid2fastq.xml` and bumping the tool version.

Pretty sure this tool is abandonware, so probably not worth the effort, I'd just tighten the `# type: ignore` here as well.

File: test/unit/webapps/test_webapp_base.py [hunk 2]
```diff
@@ -61,10 +62,10 @@ def test_parse_allowed_origin_hostnames(self):
         })
         # TODO: remove the following type ignore statement after dropping Python 3.6 support.
         # re._pattern_type has been changed to re.Pattern in python 3.7
```
Comment (nsoranzo) [inline test/unit/webapps/test_webapp_base.py]:
```suggestion
```

File: test/unit/webapps/test_request_scoped_sqlalchemy_sessions.py [hunk 3]
```diff
@@ -149,7 +150,10 @@ async def test_request_scoped_sa_session_concurrent_requests_async():
 async def test_request_scoped_sa_session_concurrent_requests_and_background_thread():
     add_request_id_middleware(app)
     # TODO: remove the following type ignore statement after dropping Python 3.6 support.
```
Comment (nsoranzo) [inline test/unit/webapps/test_request_scoped_sqlalchemy_sessions.py]:
```suggestion
```

File: packages/package.Makefile [hunk 4]
```diff
@@ -93,4 +93,4 @@ push-release:
 release: release-local push-release
 
 mypy:
-	mypy $(SOURCE_DIR) $(TEST_DIR)
+	mypy --pretty $(SOURCE_DIR) $(TEST_DIR)
```
Comment (nsoranzo) [inline packages/package.Makefile]: Should this option go to the `[mypy]` section of `setup.cfg` ?

**Comment by mr-c** [comment]
@nsoranzo Thanks for the feedback, fixed!

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**